### PR TITLE
Feature/nuc 2298 accordion component

### PR DESCRIPTION
**Pull request by iremdamla**
localhost_8080_components_accordion html

### CODE REVIEW

**Comment by Serhatkemal** [comment]
localhost_8080_components_accordion html

File: src/components/VxpAccordion.vue [hunk 1]
```diff
@@ -0,0 +1,95 @@
+<template>
+  <StackLayout class="accordion">
```
Comment (ercang) [inline src/components/VxpAccordion.vue]: I think css rule should be vxp-accordion

File: src/components/VxpAccordion.vue [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+          if (this.children_toggle_status[id] && id !== child_id) {
+            this.$emit('toggle-child', id);
+            this.onCollapsed(id);
+            this.children_toggle_status[id] = false;
+          }
+        }
+      }
+      this.children_toggle_status[child_id] = !this.children_toggle_status[child_id];
+      this.$emit('toggle-child', child_id);
+      if (this.children_toggle_status[child_id]) {
```
Comment (ercang) [inline src/components/VxpAccordion.vue]: we should follow the same naming style in all source files

File: src/components/VxpAccordionItem.vue [hunk 3]
```diff
@@ -0,0 +1,125 @@
+<template>
+  <StackLayout :class="item_class" :id="item_id">
```
Comment (ercang) [inline src/components/VxpAccordionItem.vue]: css class names should be corrected

File: src/components/VxpAccordion.vue [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      validator: items => Array.isArray(items) && !items.some(n => typeof n !== 'number'),
+    },
+    autoCollapse: {
+      required: false,
+      type: Boolean,
+      default: false,
+    },
+  },
+  mounted() {
+    this.$on('child-registered', child => {
```
Comment (osmanertem) [inline src/components/VxpAccordion.vue]: this seemed dangerous and confusing to me. I think children should trigger events like this.emit("child-registered") and this component should listen these events like <slot @child-registered>.

In current situation we listen an event that we dont know the source.

Comment (osmanertem) [inline src/components/VxpAccordion.vue]: we should avoid anonymus functions. I think it is a better implementation to keep event handlers under methods key

File: src/components/VxpAccordion.vue [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      this.children_toggle_status[child_id] = !this.children_toggle_status[child_id];
+      this.$emit('toggle-child', child_id);
+      if (this.children_toggle_status[child_id]) {
+        this.onExpanded(child_id);
+      } else {
+        this.onCollapsed(child_id);
+      }
+    },
+    openInitialItems(items_length) {
+      const is = this.initialExpandedItems;
```
Comment (osmanertem) [inline src/components/VxpAccordion.vue]: do we really need an extra variable here? if it is necessary it should be renamed with a better name

Comment (Serhatkemal) [reply src/components/VxpAccordion.vue]: renamed variables names.

File: src/components/VxpAccordion.vue (hunk 4 above)
Comment (Serhatkemal) [reply src/components/VxpAccordion.vue]: anonymus functions replaced with predefined methods.

File: src/components/VxpAccordion.vue (hunk 1 above)
Comment (Serhatkemal) [reply src/components/VxpAccordion.vue]: replaced css class names.

File: src/components/VxpAccordion.vue (hunk 2 above)
Comment (Serhatkemal) [reply src/components/VxpAccordion.vue]: replaced naming con.

File: src/components/VxpAccordionItem.vue (hunk 3 above)
Comment (Serhatkemal) [reply src/components/VxpAccordionItem.vue]: css class names updated wrt project predefined styles

**Pull request merged**